tests/data/test_minute_bars.py

```diff
@@ -451,6 +452,44 @@ def test_pad_data(self):
 
         self.assertEquals(100.0, volume_price)
 
+    def test_nans(self):
+        """
+        Test writing empty data.
+        """
+        sid = 1
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertIs(last_date, NaT)
+
+        self.writer.pad(sid, TEST_CALENDAR_START)
+
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertEqual(last_date, TEST_CALENDAR_START)
+
+        freq = self.market_opens.index.freq
+        minute = self.market_opens[TEST_CALENDAR_START + freq]
+        minutes = date_range(minute, minute + Timedelta('8 min'), freq='min')
+        data = DataFrame(
+            data={
+                'open': [nan, nan, nan, nan, nan, nan, nan, nan, nan],
```

Comment:
I did miss in that snippet above that the full is just repeating the same value, so not a thorough test.